### ISSUE TITLE
Custom Control For Import Dialogs

### ISSUE THREAD

**Issue by Patowhiz** [africanmathsinitiative/R-Instat#6153]
@rdstern @dannyparsons @lloyddewit .
Current we have several dialogs that rely on selecting and opening external files. These dialogs have similar implementations in regards to getting the file path.

I suggest that these dialogs should use custom control  similar to `ucrFileInput` which is designed for selecting file paths for exports.
 
If agreed, we could either extend the `ucrFileInput`  control to support opening of files or create a new custom control that is used specifically for opening files and writing to the R file paths parameters. Then this control can be used to replace the current `ucrInput` control and browse buttons used in the dialogs.

**Comment by rdstern** [comment]
@Patowhiz this is probably beyond my "pay-grade", but I agree of course.  I'll let @dannyparsons discuss the detail, but this could usefully be a general point, particularly for new R-Instat programmers.  My understanding is that these custom controls are central to our attempts to avoid repetition and hence should become a topic early on in the life of any new R-Instat programmer.  Perhaps we could document this example in a way it could also be useful in the future.

**Comment by Patowhiz** [comment]
@rdstern yes I agree with your point. Would be very useful to new R-Instat programmers.
